sway-core/src/ir_generation/function.rs

```diff
@@ -63,6 +66,7 @@ impl FnCompiler {
             returns_by_ref,
             recreated_fns: HashMap::new(),
             current_fn_param: None,
+            logged_types_map: logged_types_map.clone(),
```

Comment:
Does it need to be cloned?

Reply:
So yeah `logged_types_map` is a reference and changing it to be otherwise sent me down a rabbit hole. There's probably a way to do it?

Reply:
But it's a const reference, and at this stage it's never mutable again, so can't you just pass the ref around?